### PR TITLE
[Tencent Hunyuan Team] Add LoRA Inference Support for Hunyuan-DiT

### PR DESCRIPTION
**Pull request by gnobitab**
Added LoRA support to HunyuanDiT pipeline. Currently can only support `lora_scale=1`. 

You may test the PR with `test_hunyuandit_lora.py`. A pre-trained LoRA model is uploaded here: https://huggingface.co/XCLiu/hunyuandit-lora-test

Please change `YOUR_LORA_PATH` to the dir you store the downloaded lora file.

The generated image should be 
img_0

@yiyixuxu @sayakpaul Please have a look! thank you so much!

### CODE REVIEW

File: src/diffusers/models/transformers/hunyuan_transformer_2d.py [hunk 1]
```diff
@@ -179,6 +180,7 @@ def forward(
         temb: Optional[torch.Tensor] = None,
         image_rotary_emb=None,
         skip=None,
+        cross_attention_kwargs: Optional[Dict[str, Any]] = None,
```
Comment (yiyixuxu) [inline src/diffusers/models/transformers/hunyuan_transformer_2d.py:183]:
```suggestion
```

File: src/diffusers/models/transformers/hunyuan_transformer_2d.py [hunk 2]
```diff
@@ -498,13 +519,15 @@ def forward(
                     encoder_hidden_states=encoder_hidden_states,
                     image_rotary_emb=image_rotary_emb,
                     skip=skip,
+                    cross_attention_kwargs=cross_attention_kwargs,
```
Comment (yiyixuxu) [inline src/diffusers/models/transformers/hunyuan_transformer_2d.py:522]:
```suggestion
```

File: src/diffusers/models/transformers/hunyuan_transformer_2d.py [hunk 3]
```diff
@@ -498,13 +519,15 @@ def forward(
...
                 )  # (N, L, D)
             else:
                 hidden_states = block(
                     hidden_states,
                     temb=temb,
                     encoder_hidden_states=encoder_hidden_states,
                     image_rotary_emb=image_rotary_emb,
+                    cross_attention_kwargs=cross_attention_kwargs,
```
Comment (yiyixuxu) [inline src/diffusers/models/transformers/hunyuan_transformer_2d.py:530]:
```suggestion
```

File: test_hunyuan_dit_lora.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import torch
```
Comment (yiyixuxu) [inline test_hunyuan_dit_lora.py:1]: we will remove this file before merge, no?
in the future, maybe it's easier to just post testing example in PR description:)

**Review by yiyixuxu — COMMENTED** [review]
look good to me!
I think we don't need pass `cross_attention_kwargs` down if we don't use them

File: src/diffusers/models/transformers/hunyuan_transformer_2d.py (hunk 2 above)
Comment (sayakpaul) [reply src/diffusers/models/transformers/hunyuan_transformer_2d.py:522]: Agree with this!

File: test_hunyuan_dit_lora.py (hunk 4 above)
Comment (sayakpaul) [reply test_hunyuan_dit_lora.py:1]: Yes. Additionally, I think this should be turned into a proper test suite: `test_lora_layers_hunyuan_dit.py`. Just including a SLOW test is sufficient for the time being. 

Here is an example: https://github.com/huggingface/diffusers/blob/d457beed92e768af6090238962a93c4cf4792e8f/tests/lora/test_lora_layers_sd.py#L205.

File: src/diffusers/pipelines/hunyuandit/pipeline_hunyuandit.py [hunk 5]
```diff
@@ -239,6 +240,8 @@ def __init__(
             else 128
         )
 
+        self.unet_name = 'transformer' # to support load_lora_weights
```
Comment (sayakpaul) [inline src/diffusers/pipelines/hunyuandit/pipeline_hunyuandit.py:243]: Hmm, this feels a little weird to me. But okay for now. I will attempt to refactor this to harmonize. Cc: @yiyixuxu

**Review by sayakpaul — COMMENTED** [review]
Thank you for adding so quickly. Also glad that it required minimal changes on your end :-) 

I have left some comments on the implementation. 

Would it make sense to also add a note about the LoRA support (with an example) in the pipeline doc? WDYT?

**Comment by gnobitab** [comment]
Thanks for the comments! I'll update the commit.

My question is: if we don't add `cross_attention_kwargs`, how do we control `lora scale`?

I will provide a doc update after all the changes

**Comment by sayakpaul** [comment]
I think the comment was about not propagating `cross_attention_kwargs` throughout the codebase. Just the Transfomer block is fine similar to:
https://github.com/huggingface/diffusers/blob/d457beed92e768af6090238962a93c4cf4792e8f/src/diffusers/models/unets/unet_2d_condition.py#L1181

**Comment by sayakpaul** [comment]
@gnobitab please take note of https://github.com/huggingface/diffusers/pull/8670. It is going to help with this support too. IMO, it might be better to wait for a little while till that PR gets merged. Once done, we will surely revisit this.

**Pull request closed without merging**